File: src/main/java/upbrella/be/user/dto/response/AllUsersInfoResponse.java
package upbrella.be.user.dto.response;

import lombok.Builder;
import lombok.Getter;

import java.util.List;
import java.util.stream.Collectors;
import upbrella.be.user.entity.User;

@Getter
@Builder
public class AllUsersInfoResponse {

    private List<SingleUserInfoResponse> users;

    public static AllUsersInfoResponse fromUsers(List<User> users) {

        return AllUsersInfoResponse.builder()
                .users(
                        users.stream()
                                .map(SingleUserInfoResponse::fromUser)
                                .collect(Collectors.toList())
                ).build();
    }
}

File: src/main/kotlin/upbrella/be/user/dto/response/AllUsersInfoResponse.kt
package upbrella.be.user.dto.response

import upbrella.be.user.entity.User

data class AllUsersInfoResponse(val users: List<SingleUserInfoResponse>) {
    companion object {
        fun fromUsers(users: List<User?>): AllUsersInfoResponse {
            return AllUsersInfoResponse(
                users = users.map { SingleUserInfoResponse.fromUser(it!!) }
            )
        }
    }
}